src/coffee/stockimport.coffee

```diff
@@ -349,15 +350,32 @@ class StockImport
     posts = _.map inventoryEntries, (entry) =>
       existingEntry = @_match(entry, existingEntries)
       if existingEntry?
-        synced = @sync.buildActions(entry, existingEntry)
-        if synced.shouldUpdate()
-          @client.inventoryEntries.byId(synced.getUpdateId()).update(synced.getUpdatePayload())
-        else
-          Promise.resolve statusCode: 304
+        @_updateInventory(entry, existingEntry)
```

Comment:
@lojzatran Following (rather edge) cases will not be covered here:

"Delete inventory" request is send together with "update inventory" request but "delete inventory" would reach API faster. In this case we might get on "inventory update" an error of type 409 or 404. On both error types it should refetch existing inventory and calculate if it should create or update.

Reply:
https://github.com/sphereio/sphere-stock-import/issues/68